lesson-04-memory-management/mm/xxx.c

```diff
@@ -35,20 +146,37 @@ int __init x_init(void)
 {
 	int res;
 
+   pr_info("mem_type is %d\n", mem_type);
 	x_class = class_create(THIS_MODULE, "x-class");
 	if (IS_ERR(x_class))
 		pr_info("bad class create\n");
 	res = class_create_file(x_class, &class_attr_xxx);
 	pr_info("'xxx' module initialized\n");
+	init_kcache();
 	return res;
 }
 
 void x_cleanup(void)
 {
 	class_remove_file(x_class, &class_attr_xxx);
 	class_destroy(x_class);
+	deinit_kcache();
 }
 
+/**
+ * 137 * module_param_named - typesafe helper for a renamed module/cmdline parameter
+ * 138 * @name: a valid C identifier which is the parameter name.
+ * 139 * @value: the actual lvalue to alter.
+ * 140 * @type: the type of the parameter
+ * 141 * @perm: visibility in sysfs.
+ * 142 *
+ * 143 * Usually it's a good idea to have variable names and user-exposed names the
+ * 144 * same, but that's harder if the variable must be non-static or is inside a
+ * 145 * structure.  This allows exposure under a different name.
+ * 146 */
+module_param( mem_type, int, 0444 );
```

Comment:
Is it intended to be read-only?

Reply:
I made access right to read-only, because of didn't see any reasons to made other rights